### PR TITLE
Feat/validator

### PR DESCRIPTION
**Pull request by NyaGarcia**

### CODE REVIEW

File: src/validator.ts [hunk 1]
```diff
@@ -3,30 +3,41 @@ import {
...
+  const { date } = customArgs;
 
-export const validator: FieldValidationFunctionSync = fieldValidatorArgs => {
-  const { value, message = defaultMessage, customArgs } = fieldValidatorArgs;
+  if (!(value instanceof Date)) {
+    throw new TypeError(BAD_PARAMETER);
+  }
 
-  // TODO: Add validator
-  const succeeded = !isDefined(value) || ...;
```
Comment (nasdan) [inline src/validator.ts:19]: We have to keep `!isDefined(value)` because if user wants to check it, he has to use `required` validator

**Comment by nasdan** [comment]
There are some files that are pending to update. You have to find `// TODO`

File: src/validator.spec.ts [hunk 2]
```diff
@@ -1,82 +1,220 @@
...
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object earlier than customArgs date param', () => {
+      const value = new Date(2018, 11, 24, 10, 33, 30, 0);
+      const date = new Date(2018, 12, 30, 15, 33, 30, 0);
```
Comment (nasdan) [inline src/validator.spec.ts]: Be careful with this date. This date isn't `December 2018`

File: src/validator.spec.ts [hunk 3]
```diff
@@ -1,82 +1,220 @@
-import { validator, setErrorMessage } from './validator';
+import { setErrorMessage, validator } from './validator';
 
-// TODO: Add specs
 describe('fonk-previous-date-validator specs', () => {
-  it('should return succeeded validation when it feeds value equals undefined', () => {
```
Comment (nasdan) [inline src/validator.spec.ts:5]: Keep previous specs, please

File: src/validator.spec.ts [hunk 4]
```diff
@@ -1,82 +1,220 @@
...
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object with year, month, days, hours and minutes earlier than customArgs date param', () => {
+      const value = new Date(2018, 11, 24, 10, 15);
+      const date = new Date(2018, 12, 24, 10, 33);
```
Comment (nasdan) [inline src/validator.spec.ts]: Please avoid dates with month `12`  in this format. It isn't `December`

Comment (nasdan) [reply src/validator.spec.ts]: I think string format is better to get expected result:

```
const date = new Date('December 24, 2018 10:33:00');
```

File: src/validator.spec.ts [hunk 5]
```diff
@@ -1,82 +1,220 @@
...
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: false,
+        message: "Date isn't previous to the one provided.",
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should throw an error when it feeds value equals undefined', () => {
```
Comment (nasdan) [inline src/validator.spec.ts]: It doesn't have to throw an error here. See previous comment

File: src/validator.spec.ts [hunk 6]
```diff
@@ -1,82 +1,220 @@
...
-    const result = validator({ value });
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return failed validation when value is a valid Date object later than customArgs date param', () => {
+      const value = new Date(2019, 11, 24, 10, 33, 30, 0);
```
Comment (nasdan) [inline src/validator.spec.ts:136]: Do we need more failed specs?

File: src/validator.ts (hunk 1 above)
Comment (NyaGarcia) [reply src/validator.ts:19]: You want the validator to return a successful validation when passing values like undefined, null etc.?

Comment (NyaGarcia) [reply src/validator.ts:19]: Added isDefined constant and !isDefined handling => cd08fccdc089f216eec3d2ac6576651f215cb698

File: src/validator.spec.ts (hunk 5 above)
Comment (NyaGarcia) [reply src/validator.spec.ts]: Removed error Throw => cd08fccdc089f216eec3d2ac6576651f215cb698

File: src/validator.spec.ts (hunk 4 above)
Comment (NyaGarcia) [reply src/validator.spec.ts]: Changed to String format => f666b0294aa1bbded5fe45811d079f519857226f Line 137

File: src/validator.spec.ts (hunk 3 above)
Comment (NyaGarcia) [reply src/validator.spec.ts:5]: Added previous specs => f666b0294aa1bbded5fe45811d079f519857226f Lines 148 - 184

File: src/validator.spec.ts (hunk 2 above)
Comment (NyaGarcia) [reply src/validator.spec.ts]: Replaced all Dates with 12 as month => f666b0294aa1bbded5fe45811d079f519857226f

File: src/validator.spec.ts (hunk 6 above)
Comment (NyaGarcia) [reply src/validator.spec.ts:136]: Added more failed specs => c37e79cbf2c8aa224995a9d937d0a376120e1a4f

**Comment by nasdan** [comment]
Great job, thank you. I will merge it

**Pull request merged**